hoomd/md/NeighborList.h

```diff
@@ -496,11 +497,34 @@ class PYBIND11_EXPORT NeighborList : public Compute
         return m_typpair_idx;
         }
 
+    unsigned int getN() const
+        {
+        return m_pdata->getN();
+        }
+
+    unsigned int getNGhosts() const
+        {
+        return m_pdata->getNGhosts();
+        }
```

Comment:
So this is needed for an odd reason, the `LocalDataAccess` template class assumes that these methods are defined for the `Data` template parameter. The API here was weird to conform to, and I had to add an additional method `getBufferExplicitSize` since the ragged NeighborList array has a size that cannot be inferred from `getN` and `getNGhosts`.

Reply:
@b-butler is there anything we can do here? Using the local data access APIs is going to pollute all of our classes with duplicate `getN` and `getNGhosts` methods that aren't related to the class they are defined in. This shows up in `ForceCompute` already.

Reply:
The first approach is nice in that it can still handle ghost particles automatically with no additional coding by the local data classes. The second is the most general but will require more boilerplate logic. This could be ameliorated by creating helper functions that use `getBufferExplicitSize` though. In fact whatever approach we take, we ought to outsource `getBufferExplicitSize`'s logic to the other two methods.